### PR TITLE
[2.7] bpo-31754: Fix type of 'itemsize' in PyBuffer_FillContiguousStrides (GH-3993)

### PR DESCRIPTION
**Pull request by vyas45**
https://bugs.python.org/issue31754 backport

https://bugs.python.org/issue31754

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by vyas45** [comment]
@berkerpeksag here is the 2.7 backport. 

Thanks !

**Review by berkerpeksag — APPROVED** [review]
LGTM, thanks. I've just renamed the title and add "[2.7]".

**Comment by vyas45** [comment]
Thanks @berkerpeksag  for the help on the first commit. Will definitely help me with subsequent contributions.

**Comment by berkerpeksag** [comment]
> Thanks @berkerpeksag for the help on the first commit. Will definitely help me with subsequent contributions.

No problem :) Please sign the [CLA](https://www.python.org/psf/contrib/contrib-form/) for your future contributions. Otherwise, we can't accept your non-trival patches.

**Comment by vyas45** [comment]
I think i did so over the weekend. On my account I see : **Contributor Form Received** . Probably need to wait for the approval ?

**Comment by berkerpeksag** [comment]
Yes, it usually take a couple days to verify CLA forms.

**Comment by vyas45** [comment]
Got it. Will wait for the approval. 

Have a great day and week ahead !

**Pull request merged**
